### ISSUE TITLE
Private modules are not ignored

### ISSUE THREAD

**Issue by tylerhawkes** [eqrion/cbindgen#309]
This behavior is observed on cbindgen 0.8.2.
I'm trying to write a library in Rust that uses bindgen to generate some bindings. We want to export a c api using cbindgen, but I keep getting the error `thread 'main' panicked at 'not yet implemented', /rust/cargo/registry/src/github.com-1ecc6299db9ec823/cbindgen-0.8.2/src/bindgen/mangle.rs:51:37`. The bindgen code is in a module at crate::sys which is not a public module, but it has `extern "C" {pub fn...}` declarations which I think is causing the functions and everything they use to be included even though they're not public outside the crate.

I know this can be solved by using a separate crate, but it would be nice to keep everything together. Blacklisting objects isn't a great option either since there's about 10,000 lines of bindgen code. I don't see anything else in the documentation that would allow me to exclude that module for generation or to just target a single module, so having an annotation that supports that would be useful in my case, though just ignoring non-public items would fix it as well.

**Comment by RReverser** [comment]
Yeah this is important addition to visibility fixes like #314, but harder to implement in current architecture, because it requires tracking of situations like:

```rust
mod private_module;

pub use private_module::nested_module::SomeType;
```
